errors.yaml

```diff
@@ -41,7 +41,7 @@ errors:
   - code: unretrievable_document
     description: "The document exists in store, but there was an error retrieving it. This probably comes from an inconsistent state in the database."
   - code: unsupported_media_type
-    description: "The payload content type is not supported by MeiliSearch. Currently, MeiliSearch supports only JSON payloads."
+    description: "The payload content type is not supported by MeiliSearch. Currently, MeiliSearch only supports JSON, CSV, and NDJSON payloads."
```

Comment:
There are new errors described in the specs related to the format handling 🤓

- https://github.com/meilisearch/specifications/blob/develop/text/0028-indexing-csv.md#errors
- https://github.com/meilisearch/specifications/blob/develop/text/0029-indexing-ndjson.md#errors

Reply:
Absolutely @guimachiavelli! `invalid_content_type` and `missing_content_type`  and not `media_type` 😇

There are also two new `malformed_payload` and `missing_payload` errors that should be listed. I guess you were aware of it and expected to ship it with another PR?

`malformed_payload` can occur for several reasons:

- Sending a different payload type than the `Content-Type` header
- Sending an invalid CSV or NDJSON format (For now, if the CSV header does not match what we expect, the payload is considered to be invalid. We will probably have a dedicated error in the future)

`missing_payload` can occur if a payload is expected but empty

Reply:
Ha, sorry, was a bit tired when I wrote the error type names!

Are these errors parsed in the order you mentioned them here? For example, if I send an update request without `Content-Type` headers and a payload, I will get `missing_content_type` or `missing_payload`?

That aside, confirming errors and tasks:

1. remove `unsupported_media_type` ;
2. add `invalid_content_type`:  `Content-Type` header specifies an unsupported file format (e.g. SQL or zip)
3. add `missing_content_type`: payload does not contain a `Content-Type` header
4. add `malformed_payload`:  payload is not formatted according to the specified content type (i.e. user specifies e.g. JSON but the dataset is malformed or is actually something else which cannot be read as valid JSON)
5. add `missing_payload`: `Content-Type` was specified, but no file was sent to the server or the file itself is empty

Reply:
"Are these errors parsed in the order you mentioned them here? For example, if I send an update request without Content-Type headers and a payload, I will get missing_content_type or missing_payload?"

Yes, according to my point of view the `missing_content_type`error should come before `missing_payload`. I'll do some tests on the RC to see if this is the case. For the time being it's not very critical if it's not the case because the user will always find his way through the errors no matter which way.

I also confirm your last point 🤓

Reply:
updated